src/app/dashboard-widgets/create-work-item-widget/create-work-item-widget.component.spec.ts

```diff
@@ -19,90 +21,189 @@ import { WorkItemsData } from '../../shared/workitem-utils';
-    testContext.detectChanges();
+    let fakeWorkItems: WorkItem[] = [fakeWorkItem1, fakeWorkItem2];
 
-    expect(testContext.fixture.debugElement.query(By.css('#spacehome-my-workitems-add-button'))).not.toBeNull();
-    expect(testContext.fixture.debugElement.query(By.css('#spacehome-my-workitems-create-button'))).not.toBeNull();
+    let fakeWorkItemsObs: Observable<WorkItemsData> = observableOf({
+      workItems: fakeWorkItems
+    } as WorkItemsData);
 
-  });
+    type TestingContext = TestContext<CreateWorkItemWidgetComponent, HostComponent>;
+
+    const testContext: TestingContext = initContext(CreateWorkItemWidgetComponent, HostComponent, {
+      declarations: [ MockFeatureToggleComponent ],
+      imports: [NgArrayPipesModule, RouterModule, NgLetModule],
+      providers: [
+        { provide: ActivatedRoute, useValue: jasmine.createSpy('ActivatedRoute') },
+        { provide: LocationStrategy, useValue: jasmine.createSpyObj('LocationStrategy', ['prepareExternalUrl']) },
+        { provide: Broadcaster, useValue: createMock(Broadcaster) },
+        { provide: Contexts, useValue: ({ current: new Subject<Context>() }) },
+        {
+          provide: WorkItemService, useFactory: () => {
+            let workItemServiceMock = jasmine.createSpyObj('WorkItemService', ['resolveType', 'resolveAreaForWorkItem', 'getWorkItems']);
+            workItemServiceMock.resolveType.and.stub();
+            workItemServiceMock.resolveAreaForWorkItem.and.stub();
+            workItemServiceMock.getWorkItems.and.returnValue(fakeWorkItemsObs as Observable<WorkItemsData>);
+            return workItemServiceMock;
+          }
+        },
+        {
+          provide: FilterService, useFactory: () => {
+            let filterServiceMock: jasmine.SpyObj<FilterService> = jasmine.createSpyObj('FilterService', ['queryBuilder', 'queryJoiner']);
+            return filterServiceMock;
+          }
+        },
+        {
+          provide: Router, useFactory: (): jasmine.SpyObj<Router> => {
+            let mockRouterEvent: any = {
+              'id': 1,
+              'url': 'mock-url'
+            };
 
-  it('should disable buttons if the user does not own the space', function() {
-    testContext.hostComponent.userOwnsSpace = false;
-    testContext.testedDirective.myWorkItemsCount = observableOf(0);
-    testContext.detectChanges();
+            let mockRouter = jasmine.createSpyObj('Router', ['createUrlTree', 'navigate', 'serializeUrl']);
+            mockRouter.events = observableOf(mockRouterEvent);
 
-    expect(testContext.fixture.debugElement.query(By.css('#spacehome-my-workitems-add-button'))).toBeNull();
-    expect(testContext.fixture.debugElement.query(By.css('#spacehome-my-workitems-create-button'))).toBeNull();
+            return mockRouter;
+          }
+        }
+      ],
+      schemas: [
+        NO_ERRORS_SCHEMA
+      ]
+    });
+
+    it('should fetch the correct work items', async(function() {
```

Comment:
I'd rather this be `async(() => { ..` to match other similar functions throughout the codebase